README.md

```diff
@@ -28,126 +36,136 @@ Error cat was designed to be as easy as possible to use with express. Here is an
-  body: reqBody
-};
-
-// Send an http request to an external service...
-request.post(options, function (err) {
-  if (err) {
-    // Add some data about the state that caused the error
-    err.data = { body: reqBody };
-
-    // And then wrap the error so it can be logged appropriately
-    ErrorCat.wrap(err, 502);
-  }
-});
+var ErrorCat = require('error-cat')
+
+// Report an error as a warning with the given custom data
+var myError = new ErrorCat.BaseError('Something wicked...', {
+  customRollbarField: 'customValue'
+})
+myError.level = 'warn'
+ErrorCat.report(myError)
 ```
 
-#### {Boom~Error} `wrapAndReport(err, code, message)`
-Creates a new [boom](https://www.npmjs.com/package/boom) error from an existing
-error, and logs it via the `ErrorCat.log` method (see below), and reports it to
-rollbar via the `ErrorCat.report` method (see blow). Note: as with `wrap` above
-data can be supplied to the resulting boom error by setting the `.data`
-attribute on the error before wrapping it.
-
-#### {void} `log(err)`
-Logs the given error using [auto-debug](https://www.npmjs.com/package/auto-debug)
-and reports it to rollbar via `ErrorCat.report` (see below).
-
-#### {void} `report(err, [req], [cb])`
-Reports the given error via Rollbar. This method is automatically bypassed if
-`ErrorCat.canUseRollbar()` returns false (see above).
-optional req argument to add more data
-optional cb which is called after reporting
+At the top of the core error hierarchy is `BaseError`. This class has two primary
+responsibilities:
+
+1. Provide an easy-to-use and extendable error class for application programmers
+2. Set reasonable defaults for various "magic" error properties that are used
+   to control information and options when reporting errors to rollbar.
+
+Specifically, the following magic properties are set by base error:
+
+- (boolean) `.report = true` - Used to determine whether or not the error should
+  be reported to rollbar. Setting this value to `false` will cause error-cat
+  to skip reporting when encounting the error.
+- (string) `.level = 'error'` - Sets the level at which the error is reported
+  in rollbar. Acceptable values are: `critical`, `error`, `warn`, `info`.
+- (string) `.fingerprint = null` - Sets a specific fingerprint by which rollbar
+  should group the error. This is very useful for programmatically setting custom
+  groupings outside the defaults used in your rollbar configuration
+- (object) `.data = {}` - Additional custom data to be reported to rollbar. This
+  will default to an empty object literal unless it is set via the `data` argument
+  in the `BaseError` constructor.  
+
+#### class `RouteError(message, statusCode, data)` extends `BaseError`
+Used to denote exceptions that arise during the handling of RESTful routes.
+The class is automatically annotated with additional metadata via the
+[boom](https://github.com/hapijs/boom) library.
+
+#### class `TaskError(message, queue, job)` extends `BaseError`
```

Comment:
I'm having a second thought on this. Maybe this error should belong to Ponos repo? It's hard to see how it would be used outside of the Ponos.
 Just an opinion and non-blocker